submission/persistence-common-api/src/main/kotlin/ac/uk/ebi/biostd/persistence/common/service/SubmissionOperations.kt

```diff
@@ -55,14 +47,38 @@ interface SubmissionPersistenceQueryService {
      * @param filter the submission filter
      **/
     fun getSubmissionsByUser(owner: String, filter: SubmissionFilter): List<BasicSubmission>
+}
+
+interface SubmissionRequestPersistenceService {
+    fun hasActiveRequest(accNo: String): Boolean
+
+    fun saveSubmissionRequest(rqt: SubmissionRequest): Pair<String, Int>
+
+    fun createSubmissionRequest(rqt: SubmissionRequest): Pair<String, Int>
+
+    fun updateRequestStatus(accNo: String, version: Int, status: RequestStatus)
 
     fun getPendingRequest(accNo: String, version: Int): SubmissionRequest
+
     fun getLoadedRequest(accNo: String, version: Int): SubmissionRequest
+
     fun getCleanedRequest(accNo: String, version: Int): SubmissionRequest
-    fun getReferencedFiles(accNo: String, fileListName: String): List<ExtFile>
+
     fun getRequestStatus(accNo: String, version: Int): RequestStatus
 }
 
+interface SubmissionFilesPersistenceService {
+    fun saveSubmissionFile(file: SubmissionFile)
+
+    fun getSubmissionFile(path: String, accNo: String, version: Int): ExtFile
+
+    fun getSubmissionFiles(accNo: String, version: Int, startingAt: Int = 0): List<Pair<ExtFile, Int>>
+
+    fun getFileListFiles(accNo: String, version: Int, fileListName: String): List<ExtFile>
```

Comment:
Not sure if this is temporally but this operation need to be paginated or streamed. Loading all files in memory has been proved to  not be a good option.  We may use spring java doc stream feature https://docs.spring.io/spring-data/mongodb/docs/current/reference/html/#repositories.query-streaming.

Reply:
Yes, the idea is to either make it a stream or a paginated one in the next (and hopefully last PR)